### ISSUE TITLE
wasm platform benchmark

### ISSUE THREAD

**Issue by jfmc** [jprendes/emception#1]
First of all, congrats for this awesome project!

It feels (I've not measured it) too slow compared with native emscripten. Have you considered pushing it as a (real world) wasm platform (and emscripten) benchmark? I believe that a detailed performance analysis will show very interesting data for wasm compilers and wasm-based platforms (file system simulation, etc.).

**Comment by Froggo8311** [comment]
Yeah... It seems to be taking a while to load and then compile...

**Comment by alienself** [comment]
Having the same issue, what is causing this slow down compared to native?

**Comment by windowsair** [comment]
I think there are tons of system API calls that are causing the slowness. These system calls require the intervention of the js runtime.  Maybe WASI is a good idea?

IO bottlenecks, multithreading, and memory allocators are also important to consider.
